server/routes/ad.js

```diff
@@ -19,60 +19,73 @@ router.put('/', (req, res, next) => {
     email,
     zipCode,
   } = req.body;
-  sheets.spreadsheets.values.get(
-    {
-      auth: jwtClient,
-      spreadsheetId: spreadsheetId,
-      range: sheetRange,
-    },
-    (err, response) => {
-      if (err) {
-        console.log(err);
-      } else {
-        console.log('Ad retrieved from ' + sheetRange);
-        const ads = response.data.values;
-        let rowCounter = 2;
-        for (let ad of ads) {
-          if (ad[0] === req.body.id) {
-            let values = [
-              [
-                req.body.id,
-                title,
-                price,
-                description,
-                photo,
-                condition,
-                email,
-                zipCode,
-                modifiedDate,
-              ],
-            ];
-            const resource = { values };
-            return sheets.spreadsheets.values.update(
-              {
-                auth: jwtClient,
-                spreadsheetId: spreadsheetId,
-                range: `Ads!A${rowCounter}:J${rowCounter}`,
-                valueInputOption: 'USER_ENTERED',
-                resource,
-              },
-              (err, response) => {
-                if (err) {
-                  console.log(err);
-                } else {
-                  console.log('ad');
-                  const ads = response.data.values;
-                  return res.status(204).json();
-                }
-              }
-            );
-          }
-          rowCounter++;
-        }
-        return res.status(404).json('Ad Not Found. Please check ID.');
+
+  const sheetsReq = {
+    auth: jwtClient,
```

Comment:
@raymond-sde , were you saying that each time we pass sheetsReq with the jwtClient, we're generating a new access token for each request?